sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4223,6 +4223,17 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     checkAnswer(sql("""SELECT from_json(r'{"a": "\\"}', 'a string')"""), Row(Row("\\")))
     checkAnswer(sql("""SELECT from_json(R'{"a": "\\"}', 'a string')"""), Row(Row("\\")))
   }
+
+  test("SPARK-36851: Incorrect parsing of negative ANSI typed interval literals") {
+    val df1 = sql("select interval -'1' year")
```

Comment:
Let's move the test cases into `interval.sql`.
Also, we should add tests for negative day time intervals as well.

Reply:
OK

Reply:
done